src/pipeline.ts

```diff
@@ -194,6 +194,19 @@ export class GitHubWorkflow extends PipelineBase {
     // eslint-disable-next-line no-console
     console.error(`writing ${this.workflowPath}`);
     mkdirSync(path.dirname(this.workflowPath), { recursive: true });
+
+    if (process.env.SYNTH === 'github') {
+      // check if workflow file has changed
+      if (!existsSync(this.workflowPath)) {
+        throw new Error('The committed workflow file differs from the synthesized workflow file. Did you forget to commit?');
```

Comment:
You can get rid of the duplicated error message. Either that, or make this one more accurate.

Reply:
I'm debating whether "Did you forget..." sounds accusatory 😅.

How about "please commit the updated workflow file (${relative path here}) when you change your pipeline definition."